### ISSUE TITLE
[idea] Mobile UI #4: Dive list different layouts and menu option

### ISSUE THREAD

**Issue by Bocio** [subsurface/subsurface#428]
Previous idea here: 
[Mobile UI #1: general layout](https://github.com/Subsurface-divelog/subsurface/issues/425)
[Mobile UI #2: default action for main screen button](https://github.com/Subsurface-divelog/subsurface/issues/426)
[Mobile UI #3: Dive list item layout](https://github.com/Subsurface-divelog/subsurface/issues/427)

All original mockups made with moqups and are online at: (https://app.moqups.com/Bocio/3SSyFCoEwP)

## Default dive list and menu option
I show here an idea for the main menu option:

![subsurface-mobile-main-options](https://user-images.githubusercontent.com/27133591/26927634-fe36b164-4c52-11e7-9632-b5b52dcd9326.JPG)

It has six features:
- default dive list
- photo dive list
- featured dive list
- refresh aka cloud syncronize
- sort by date (current implementation)
- sort by divesite (more on this later)

## Default dive list
We already saw the default dive list. It has a fancy icon indicating dive type.

![subsurface-mobile-main](https://user-images.githubusercontent.com/27133591/26927783-8e236ace-4c53-11e7-9356-b5e80a3c5579.JPG)

## Photo dive list

@atdotde put a great effort on enabling photos management inside Subsurface and while it's something I don't care I see that there are a lot of people who love it. So now we have one or more photos attached to one dive. We could leverage this feature to have a dive list with one photo as thumbnail, Who don't use photos and choose list/magazine view will get a default thumbnail/image. It could be Subsurface logo, or a nice picture for each dive type (an actual photo not an icon). I can give you some CC licensed photos to use. A reb diver, a oc diver, a free diving diver etc...
Here an example:

![subsurface-mobile-main-list-image](https://user-images.githubusercontent.com/27133591/26927801-9f094818-4c53-11e7-965f-a144662a0cd4.JPG)

## Featured dive list

Here I copied the idea from Wordpress and Feedly mobile app. Basically once we have one or more photo linked to a dive, we could use one of them to create one cover for each dive. Please note that our current scroll method would not be ok for this kind of feature. We would need a page by page scrolling (I will add an explicative video later). We could just pick only certain dives for this kind of visualization i.e. only dives with 5 start rating...
Again, if we haven't photos attached we will get a default picture as previous example.
This is an example of how it could be implemented (I added some bells and whistles):

![subsurface-mobile-main-featured-dive](https://user-images.githubusercontent.com/27133591/26928053-cdada2f8-4c54-11e7-953e-45b3f76f1a22.JPG)

**Comment by dirkhh** [comment]
This issue has gorgeous ideas in it. I'd love to see some of this be implemented. I don't think I have the skill (or time) to tackle this, but year, I love this.

**Comment by Bocio** [comment]
> This issue has gorgeous ideas in it. I'd love to see some of this be implemented. I don't think I have the skill (or time) to tackle this, but year, I love this.

Yes, everything is pretty much explained. We need "just" a developer on it.

**Comment by dirkhh** [comment]
and once again I'll just name tag @jbygdell since @neolit123 forgot to add him.
This may be the killer feature for Subsurface-mobile 3.0 as it needs more infrastructure on the git side and the desktop version (we need to define this "one picture per dive" and also show this somehow on the desktop to make things consistent and useful, I think)

Maybe after 4.7 and 2.0 are stable, ready on all platforms, and released, we could tackle that.
